packages/babel-helpers/src/helpers/applyDecs.js

```diff
@@ -14,11 +14,24 @@
   SETTER = 4;
 
   STATIC = 5;
+
+  CLASS = 10; // only used in assertValidReturnValue
 */
 
-function createMetadataMethodsForProperty(metadataMap, kind, property) {
+function createMetadataMethodsForProperty(
+  metadataMap,
+  kind,
+  property,
+  decorationState
+) {
   return {
     getMetadata: function (key) {
+      if (decorationState.finished) {
```

Comment:
Nit: If `decoratorState` only contains `finished`, they can be mangled as `decoratorFinished` with `v` or whatever.

Reply:
It needs to be an object because it's a "live reference" (it's updated in the caller). We can rename it `decoratorFinishedRef = { v: true }` to make it more clear and smaller.